Artificer Expansion Pack: Contraptions

Nakalon opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->

## About The Pull Request

This Expansion Pack (29.99$) includes two new toys!
CONPTRAPTIONS!

Now Artificers get to make a machine and experiment with it!

These contraptions require something to power them, which wastes away after X amount of uses (defined by contraption).
In what you'll see in these videos it them taking cogs to function. This will probably change in the future at least for the smelter (it will accept coal instead).

These machines have a chance to misfire, as you will see in a video down there, this chance is reduced with engineering skill and good luck!

The idea here is to make machines which have creative uses, that are slightly comical, fun to play around with and are unpredictable.
They hopefully reward experimentation having very different outcomes and things that may happen to them!
A very important part of this is the feel and sound of them. Which I took some care with!

They were made with future machines in mind, making it easy to code more!

Presenting the contraptions:

***At present I haven't made sprites for them yet!***

### Portable Smelter

This handy device will turn things into other things.
Yeah, it will smelt them, go figure.

***This contraption has 6 uses and a misfire chance of 10.***

https://www.youtube.com/watch?v=YJiwSwH09yg&feature=youtu.be

I've gone the extra mile and added smelt results to some other unusual items, where it would make sense.

### Wood Metalizer

This one is funky.
Its a machine that turns wood into metal. What the fuuuuuuuuuck?!

***This contraption has 5 uses (in the video you see 3) and a misfire chance of 10.***

https://www.youtube.com/watch?v=0X1RaJEx6sc

#### Now, its important to say these machines have a chance to fail in a comical way.

Here's me fucking around and finding out:

https://www.youtube.com/watch?v=BcHQlUjmJSs

## Why It's Good For The Game

I was asked like 3 times what the Artificer was about after making the last PR, and honestly, I couldn't give a straight answer.
My vision is to make the artificers have wacky insane machines like these that reward experimentation.

The theme of this pack is metal because that's the other half of Artificing that I didn't really touch and is a good way to ensure artificers and smiths love each other in a weird way.

## Changelog

add: Added two contraptions, the portable smelter and the wood metalizer
tweak: Some items were previously unable to be clicked or attacked. Now they can. (Think of sticks)
code: Moved the smeltresult var from obj/item to obj

Nakalon commented:
last round of testing

Keazra commented:
This is rad. Cool vibes, honestly. The failure chance is amusing.

Now we're one step closer to automated fucking machines..

Nakalon commented:
> This is rad. Cool vibes, honestly. The failure chance is amusing.
> 
> Now we're one step closer to automated fucking machines..

I wanted to make the failure chance funny!
My theme for these is creativity, unpredictability and comedic timing!

I have 2 more planed and some improvements to these ones too!

Nakalon commented:
I fixed some street lamp sprites in this commit because I wanted to use those (in this PR, not telling where not to break the fun of discovering)

Door lock issue fixed.

Added more smeltable and metalized results.

can't remember what else

All bugs found solved but:
Fix power supply disappearing // NOT FIXED

Nakalon commented:
AH, also fixed the "unusable quiver" bug caused by this by setting object flag on quivers back to 0

Nakalon commented:
![image](https://github.com/user-attachments/assets/f5723ebd-b98b-4275-bf36-c71eeb7f1927)

Fixed in next commit

Nakalon commented:
Resolved all bugs!

In the end:

Removed object_flags when that var was duplicated (it is present in obj/item)

Fixed the issue where the contraptions would lose their charges randomly.

Modified the misfire of furnace.

Changed some smelt results which were unjustly weak.

Gyrandola commented on code/modules/roguetown/roguejobs/artificer/contraptions.dm:
```diff
@@ -0,0 +1,235 @@
+		S.set_up(1, 1, front)
+		S.start()
+		return
+	else
+		var/skill = user.mind.get_skill_level(/datum/skill/craft/engineering)
+		if(istype(O, /obj/structure/mineral_door/wood)) //This is to ensure the new door will retain its lock
+			var/obj/structure/mineral_door/wood/I = O
+			var/obj/structure/mineral_door/wood/new_door = new I.metalizer_result(get_turf(I))
+			new_door.locked = I.locked
+			if(I.lockid)
+				new_door.lockid = I.lockid
+			qdel(I)
+		else
+			var/obj/I = O
+			new I.metalizer_result(get_turf(I))
+			qdel(I)
+		flick(on_icon, src)
+		current_charge -= 1
+		shake_camera(user, 1, 1)
+		playsound(src, 'sound/magic/swap.ogg', 100, TRUE)
+		user.mind.add_sleep_experience(/datum/skill/craft/engineering, (user.STAINT / 2))
+		if(misfire_chance && prob(max(0, misfire_chance - user.goodluck(2) - skill)))
+			misfire(O, user)
+		if(!current_charge)
+			addtimer(CALLBACK(src, PROC_REF(battery_collapse), O, user), rand(5))
+		return
```
the first ELSE and subsequent indentation is redundant

Nakalon replied:
Okay ball buster 😡
